mvpa2/algorithms/searchlight_hyperalignment.py

```diff
@@ -143,6 +147,13 @@ def _call(self, ds):
                         fs[seed_index] = max(fs)
                     features_selected.append(fselector(fs))
                 ds = [sd[:, fsel] for fsel, sd in zip(features_selected, ds)]
+        elif np.sum(nfeatures_bin) > 0:
+            feature_scores = compute_feature_scores(ds, self.exclude_from_model)
+            feature_scores = [feature_scores[isub] for isub in np.where(nfeatures_bin)[0]]
+            fselector = FixedNElementTailSelector(nfeatures, tail='upper', mode='select', sort=False)
+            features_selected = [fselector(fs) for fs in feature_scores]
+            for isub_large, isub in enumerate(np.where(nfeatures_bin)[0]):
```

Comment:
here what you actually wanted  is `for selected_features, isub in zip(features_selected, bigger_ds_idxs):  ds[isub] = ds[isub][:, selected_features]`.  avoids unnecessary indexing, again -- somewhat easier to read by avoiding enumerate -- makes immediate association between two lists to be traversed